src/typescript/sdk/tests/utils/generate-mock-data.ts

```diff
@@ -0,0 +1,316 @@
+      emojicoin: emojicoins[0],
+      isSell: true,
+      inputAmount: 14000000000000000n,
+    },
+  });
+  actions.push({
+    type: "swap",
+    data: {
+      account: accounts[1],
+      emojicoin: emojicoins[0],
+      isSell: true,
+      inputAmount: 14000000000000000n,
+    },
+  });
+  actions.push({
+    type: "swap",
+    data: {
+      account: accounts[2],
+      emojicoin: emojicoins[0],
+      isSell: true,
+      inputAmount: 14000000000000000n,
+    },
+  });
+
+  // Creating whale (~42.17% of supply)
+  actions.push({
+    type: "swap",
+    data: {
+      account: accounts[3],
+      emojicoin: emojicoins[0],
+      isSell: false,
+      inputAmount: ONE_APTN * 50000n,
+    },
+  });
+
+  // Creating dolphin (~3.12% of supply)
+  actions.push({
+    type: "swap",
+    data: {
+      account: accounts[4],
+      emojicoin: emojicoins[0],
+      isSell: false,
+      inputAmount: ONE_APTN * 200000n,
+    },
+  });
+
+  // Creating puffer fish (~0.0003% of supply)
+  actions.push({
+    type: "swap",
+    data: {
+      account: accounts[5],
+      emojicoin: emojicoins[0],
+      isSell: false,
+      inputAmount: ONE_APTN * 100n,
+    },
+  });
+
+  // Provide and remove some liquidity
+  for (let i = 0n; i < 10n; i += 1n) {
+    actions.push({
+      type: "provideLiquidity",
+      data: {
+        account: accounts[3],
+        emojicoin: emojicoins[0],
+        quoteAmount: (i + (i % 2n === 0n ? 1000000n : 500000n)) * ONE_APTN,
+      },
+    });
+    actions.push({
+      type: "removeLiquidity",
+      data: {
+        account: accounts[3],
+        emojicoin: emojicoins[0],
+        lpCoinAmount: (i + (i % 2n === 0n ? 50n : 100n)) * ONE_APTN,
+      },
+    });
+  }
+
+  // Make each account chat once on each market
+  for (const emojicoin of emojicoins) {
+    for (const account of accounts) {
+      actions.push({
+        type: "chat",
+        data: {
+          account,
+          emojicoin,
+          emojiBytes: ["f09f8f81"],
+          emojiIndicesSequence: new Uint8Array([0]),
+        },
+      });
+    }
+  }
+
+  await execute(aptos, actions);
+};
+
+const getTypeTags = (lazy: Lazy<EmojicoinInfo>): [TypeTagInput, TypeTagInput] => [
+  lazy.get().emojicoin,
+  lazy.get().emojicoinLP,
+];
+
+async function execute(aptos: Aptos, actions: Action[]) {
+  const defaultTx = {
+    aptosConfig: aptos.config,
+    options: {
+      maxGasAmount: ONE_APT / 100,
+      gasUnitPrice: 100,
+    },
+    integrator: Account.generate().accountAddress,
+  };
+  const defaultTxWithMarket = (emojicoin: Lazy<EmojicoinInfo>) => ({
+    aptosConfig: aptos.config,
+    marketAddress: emojicoin.get().marketAddress,
+    typeTags: getTypeTags(emojicoin),
+    options: {
+      maxGasAmount: ONE_APT / 100,
+      gasUnitPrice: 100,
+    },
+    integrator: Account.generate().accountAddress,
+  });
+  for (const action of actions) {
+    if (action.type === "registerMarket") {
+      /* eslint-disable-next-line no-await-in-loop */
```

Comment:
You may already know this, but in case you don't, you can always disable this on a per-file basis at the top of the file 😄

```typescript
/* eslint-disable no-await-in-loop */
import { Account, type HexInput, type Aptos, type Uint64, Hex } from "@aptos-labs/ts-sdk";
// ...
// ...
```